pkg/controller/nodelifecycle/node_lifecycle_controller_test.go

```diff
@@ -92,11 +92,10 @@ func (nc *nodeLifecycleController) doEviction(fakeNodeHandler *testutil.FakeNode
 	defer nc.evictorLock.Unlock()
 	zones := testutil.GetZones(fakeNodeHandler)
 	for _, zone := range zones {
-		nc.zonePodEvictor[zone].Try(func(value scheduler.TimedValue) (bool, time.Duration) {
```

Comment:
what uses this `nodeLifecycleController`?

Shouldn't those tests be removed instead?

Reply:
It's now used by `TestMonitorNodeHealthEvictPodsWithDisruption`, which covers the testcases about method `monitorNodeHealth`.

It's just a fake eviction, has very little to do with taint based eviction or non-taint based eviction. I think we should keep them to cover the cases.

Reply:
But it's not something that happens in production anymore, so the test should change.

IIUC, the effect should be that once the node is found not healthy, the controller adds a NoExecute taint. Although I suspect that's already tested somewhere else.